File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        var x = {};
        for (var i = 0; i < numbers.length; i++) {
            var complement = target - numbers[i];
            if (x[complement] != undefined) {
                return [x[complement], i + 1];
            }
            x[numbers[i]] = i + 1;
        }
        return []
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();  
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (map.get(complement) != null){
                int[] list = new int[2];
                list[0] = map.get(complement);
                list[1] = i;
                return list;
            }
            map.put(nums[i], i);
        }
        return new int[0];
    }
}
